Extension/src/LanguageServer/protocolFilter.ts

```diff
@@ -7,30 +7,60 @@
 import { Middleware } from 'vscode-languageclient';
 import { ClientCollection } from './clientCollection';
 import { Client } from './client';
+import { SourceFileConfiguration, CustomConfigurationProvider } from '../api';
+import { provideCustomConfiguration } from './extension';
+
+function runThenableWithTimeout(promise: Thenable<any>, ms: number): Thenable<any> {
```

Comment:
>runThenableWithTimeout [](start = 9, length = 22)

btw general comment: might be nice to have a utils.ts for the extension with common functions such as this being co-located. 

Not blocking for this PR